### DISCUSSION TITLE
Copilot nerfed beyond usability

### DISCUSSION THREAD

**Discussion by omarabid** [community/community#60116] [Copilot Conversations]
### Select Topic Area

Product Feedback

### Body

Copilot is now nerfed beyond usability (am mostly programming in Rust). It's worse now than at the point when it was released.

To give one small example, when I am in Cargo.toml and I start adding a dependency, Copilot usually auto complete that with an older version (reasonable since it was cut off in the past). The problem however, is that it adds "```" at the end. This suggest that it think that it is in some kind of mark down document. Which it is not.

[dependencies]
ra+nd = "0.8.3"```+

This might not be such a big deal except that now for most of the code it is generating it's mostly hallucinations. If I ask for something, it simply makes up a "crate" out of think air and invest a function name that should solve my problem.

**Comment by andreujuanc** [comment]
100%. I'm not happy with it.

**Comment by Kevin759628** [comment]
Copilot was nerfed 100%. I have browser games I developed with bing before it became copilot and that is proof of how much better it was. I can't create like I was with bing. Copilot forgets the context of the conversation on the very next message. I knew nothing of coding, I learned to build things like depth points which are like z-index layers so ym players and npcs could walk around objects, I modulated my elements and used coordinates. I used global alphas and collisons. Image paths. I used doors to connect areas. Now all I have is what bing helped me do but nothing nearly as elaborate can happen anymore. =^( Please bring back the old bing version which was better. Somone stole my sense of accomplishment and production ability when they nerfed bing now known as copilot.

**Reply by ayim** [reply]
Which IDE are you in?

**Comment by jfullstackdev** [comment]
as for me, nope, if you know how to use the tool, then you will reap the benefits

remember that it does not understand fully the human context, that's why it's still a tool that can't replace humans

the best part for me is having suggestions, ideas, and even high quality code, I use the Copilot Chat more than the in-line Editor as I can ask the Copilot Chat for specific questions, and I feel like I matured in coding for 5 years

it's very efficient in coding too, as when you know the correct code even by just reading, you can simply ask Copilot to do it, it also can generate great documentation, unit tests
